### PR TITLE
clear template menu before appending new templates

### PR DESCRIPTION
**Pull request by pineapplespatula**
fix for issue #1836

### CODE REVIEW

File: src/js/models/menubar.js [hunk 1]
```diff
@@ -238,6 +238,7 @@ define(function (require, exports, module) {
             templateMenu = fileMenu.submenu[templateIndex],
             templateSubmenu = templateMenu.submenu;
 
+        templateSubmenu.length = 0;
```
Comment (volfied) [inline src/js/models/menubar.js]: Why not just start templateSubmenu as an empty array?

**Comment by volfied** [comment]
Got a nit against redundancy of the redundancy, @pineapplespatula

File: src/js/models/menubar.js (hunk 1 above)
Comment (pineapplespatula) [inline src/js/models/menubar.js]: because we need to make sure we clear the current submenu in this case to avoid duplicate entries.

Comment (iwehrman) [inline src/js/models/menubar.js]: It would be less scary to do something like `templateMenu.submenu = templates.map(...)`. Can you try that instead?

**Comment by pineapplespatula** [comment]
@iwehrman it is prettier now, so back to you

**Comment by iwehrman** [comment]
Brilliant.

**Pull request merged**